runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java

```diff
@@ -40,6 +47,7 @@
     private final MetricFilter metricFilter;
     private PrometheusReporter prometheusReporter;
     private CollectorRegistry collectorRegistry;
+    private SampleBuilder sampleBuilder;
```

Comment:
Any reason for this to be a class variable? It seems to be used only in start method

Reply:
The above requested change is addressed by the commit 2523d00.